internal/controller/state/graph/secret.go

```diff
@@ -72,14 +87,33 @@ func (r *secretResolver) resolve(nsname types.NamespacedName) error {
 		}
 
 		certBundle = NewCertificateBundle(nsname, "Secret", cert)
-	}
 
-	r.resolvedSecrets[nsname] = &secretEntry{
-		Secret: Secret{
-			Source:     secret,
-			CertBundle: certBundle,
-		},
-		err: validationErr,
+		r.resolvedSecrets[nsname] = &secretEntry{
+			Secret: Secret{
+				Source:     secret,
+				CertBundle: certBundle,
+			},
+			err: validationErr,
+		}
+	case secret.Type == apiv1.SecretType(SecretTypeHtpasswd):
+		// Validate Htpasswd secret
+		if _, exists := secret.Data[AuthKeyBasic]; !exists {
+			validationErr = fmt.Errorf("missing required key %q in secret type %q", AuthKeyBasic, secret.Type)
+		}
+		r.resolvedSecrets[nsname] = &secretEntry{
+			Secret: Secret{
+				Source: secret,
+			},
+			err: validationErr,
+		}
+	default:
+		validationErr = fmt.Errorf("unsupported secret type %q", secret.Type)
+		r.resolvedSecrets[nsname] = &secretEntry{
+			Secret: Secret{
+				Source: secret,
+			},
+			err: validationErr,
+		}
```

Comment:
I think we can reduce a bit of duplication in this function.

The common pieces could all be done at the end. It looks like the `certBundle` field is the only unique part, so that could be the one difference.

Reply:
Sounds good. I'll see what I can do.

Reply:
You pointed out correctly [here](https://github.com/nginx/nginx-gateway-fabric/pull/4436#discussion_r2608407101) that it doesn't make sense for the Basic Auth case.
The way this code was previously looks to be the only way to accomplish this. 

It would just mean the value for CertBundle would be `nil` for any situation that isn't the TLS case.